### ISSUE TITLE
receiving "code 18: self signed certificate" for secondary master in HA config

### ISSUE THREAD

**Issue by bl4kd3th** [Icinga/icinga2#5923]
## Expected Behavior
I expect it to throw that error in the log until I run the ```icinga2 ca sign``` command on the primary master on all master/satellite nodes.

## Current Behavior
I continue to receive that error message in the primary master's ```/var/log/icinga2/icinga2.log``` for the secondary master even after running the command. The satellites begin working after signing their certs.

## Steps to Reproduce (for bugs)
1. use the node wizard to setup primary master as master node, and secondary master and 2 satellites as secondary nodes
2. Update ```zones.conf``` on the nodes

Primary Master (icinga2-master1.localdomain) config:
```
object Endpoint NodeName {
}

object Endpoint "icinga2-master2.localdomain" {
        host = "icinga2-master2.localdomain"
}

object Endpoint "icinga2-satellite1.localdomain" {
        host = "icinga2-satellite1.localdomain"
}

object Endpoint "icinga2-satellite2.localdomain" {
        host = "icinga2-satellite2.localdomain"
}

object Zone "master" {
        endpoints = [ NodeName, "icinga2-master2.localdomain" ]
}

object Zone "west" {
        endpoints = [ "icinga2-satellite1.localdomain" ]
        parent = "master"
}

object Zone "east" {
        endpoints = [ "icinga2-satellite2.localdomain" ]
        parent = "master"
}

object Zone "global-templates" {
        global = true
}

object Zone "director-global" {
        global = true
}
```
Secondary Master (icinga2-master2.localdomain) config:
```
object Endpoint NodeName {
}

object Endpoint "icinga2-master1.localdomain" {
        host = "icinga2-master1.localdomain"
}

object Endpoint "icinga2-satellite1.localdomain" {
        host = "icinga2-satellite1.localdomain"
}

object Endpoint "icinga2-satellite2.localdomain" {
        host = "icinga2-satellite2.localdomain"
}

object Zone "master" {
        endpoints = [ NodeName, "icinga2-master1.localdomain" ]
}

object Zone "west" {
        endpoints = [ "icinga2-satellite1.localdomain" ]
        parent = "master"
}

object Zone "east" {
        endpoints = [ "icinga2-satellite2.localdomain" ]
        parent = "master"
}

object Zone "global-templates" {
        global = true
}

object Zone "director-global" {
        global = true
}
```
Satellite 1 config:
```
object Endpoint NodeName {
}

object Endpoint "icinga2-master2.localdomain" {
}

object Endpoint "icinga2-master1.localdomain" {
}


object Zone "master" {
        endpoints = [ "icinga2-master1.localdomain", "icinga2-master2.localdomain" ]
}

object Zone "west" {
        endpoints = [ NodeName ]
        parent = "master"
}


object Zone "global-templates" {
        global = true
}

object Zone "director-global" {
        global = true
}
```
Satellite 2 config:
```
object Endpoint NodeName {
}

object Endpoint "icinga2-master2.localdomain" {
}

object Endpoint "icinga2-master1.localdomain" {
}


object Zone "master" {
        endpoints = [ "icinga2-master1.localdomain", "icinga2-master2.localdomain" ]
}

object Zone "east" {
        endpoints = [ NodeName ]
        parent = "master"
}


object Zone "global-templates" {
        global = true
}

object Zone "director-global" {
        global = true
}
```
3. Your icinga2.log on primary master should have entries like these:

for satellites
```
[2017-12-29 10:58:03 -0600] warning/ApiListener: Certificate validation failed for endpoint 'icinga2-satellite1.localdomain': code 18: self signed certificate
Context:
        (0) Handling new API client connection

[2017-12-29 10:58:03 -0600] information/ApiListener: New client connection for identity 'icinga2-satellite1.localdomain' to [10.12.1.2]:5665 (certificate validation failed: code 18: self signed certificate)
[2017-12-29 10:58:03 -0600] information/ApiListener: Finished reconnecting to endpoint 'icinga2-satellite1.localdomain' via host 'icinga2-satellite1.localdomain' and port '5665'
[2017-12-29 10:58:03 -0600] information/JsonRpcConnection: Received certificate request for CN 'icinga2-satellite1.localdomain' not signed by our CA.
[2017-12-29 10:58:03 -0600] information/JsonRpcConnection: Certificate request for CN 'icinga2-satellite1.localdomain' is pending. Waiting for approval.
```
for secondary master
```
[2017-12-29 10:58:03 -0600] warning/ApiListener: Certificate validation failed for endpoint 'icinga2-master2.localdomain': code 18: self signed certificate
Context:
        (0) Handling new API client connection

[2017-12-29 10:58:03 -0600] information/ApiListener: New client connection for identity 'icinga2-master2.localdomain' to [10.11.1.2]:5665 (certificate validation failed: code 18: self signed certificate)
[2017-12-29 10:58:03 -0600] information/ApiListener: Finished reconnecting to endpoint 'icinga2-master2.localdomain' via host 'icinga2-master2.localdomain' and port '5665'
```
4. Issue the ```icinga2 ca list``` command on primary master, and then sign each of the requests (the secondary master and 2 satellites)
5. After a few minutes, your icinga2.log on primary master should now have entries like these:

for satellites
```
[2017-12-29 11:00:03 -0600] information/ApiListener: Reconnecting to endpoint 'icinga2-satellite1.localdomain' via host 'icinga2-satellite1.localdomain' and port '5665'
[2017-12-29 11:00:03 -0600] information/ApiListener: New client connection for identity 'icinga2-satellite1.localdomain' to [10.12.1.2]:5665
[2017-12-29 11:00:03 -0600] information/ApiListener: Finished reconnecting to endpoint 'icinga2-satellite1.localdomain' via host 'icinga2-satellite1.localdomain' and port '5665'
[2017-12-29 11:00:03 -0600] information/JsonRpcConnection: Received certificate request for CN 'icinga2-satellite1.localdomain' signed by our CA.
```
for secondary master
```
[2017-12-29 11:00:03 -0600] warning/ApiListener: Certificate validation failed for endpoint 'icinga2-master2.localdomain': code 18: self signed certificate
Context:
        (0) Handling new API client connection

[2017-12-29 11:00:03 -0600] information/ApiListener: New client connection for identity 'icinga2-master2.localdomain' to [10.11.1.2]:5665 (certificate validation failed: code 18: self signed certificate)
[2017-12-29 11:00:03 -0600] information/ApiListener: Finished reconnecting to endpoint 'icinga2-master2.localdomain' via host 'icinga2-master2.localdomain' and port '5665'
```
6. at this point commands aren't being load balanced between the 2 master zones.

## Context
I set up the 4 nodes before the 2.8 update and wanted to rename them. I went through the same process as before (following the documentation) and was unable to get the HA masters to communicate. Only after manually signing the cert on the secondary master did they start splitting checks.

To manually sign, I ran these commands on the secondary master:
```
# cd /var/lib/icinga2/certs/
# icinga2 pki new-cert --cn icinga2-master2.localdomain --key icinga2-master2.localdomain.key --csr icinga2-master2.localdomain.csr
# icinga2 pki sign-csr --csr icinga2-master2.localdomain.csr --cert icinga2-master2.localdomain.crt
```

## Your Environment
* Version used (`icinga2 --version`): r2.8.0-1
* Operating System and version: CentOS Linux release 7.4.1708 (Core)
* Enabled features (`icinga2 feature list`): api checker graphite ido-mysql mainlog notification (on both masters)
* Icinga Web 2 version and modules (System - About): 2.5.0, director 1.4.2
* Config validation (`icinga2 daemon -C`):
```
information/cli: Icinga application loader (version: r2.8.0-1)
information/cli: Loading configuration file(s).
information/ConfigItem: Committing config item(s).
information/ApiListener: My API identity: icinga2-master2.localdomain
warning/ApplyRule: Apply rule 'mail-icingaadmin' (in /etc/icinga2/conf.d/notifications.conf: 11:1-11:45) for type 'Notification' does not match anywhere!
warning/ApplyRule: Apply rule 'mail-icingaadmin' (in /etc/icinga2/conf.d/notifications.conf: 23:1-23:48) for type 'Notification' does not match anywhere!
warning/ApplyRule: Apply rule 'backup-downtime' (in /etc/icinga2/conf.d/downtimes.conf: 5:1-5:52) for type 'ScheduledDowntime' does not match anywhere!
warning/ApplyRule: Apply rule 'cluster-zone' (in /var/lib/icinga2/api/zones/director-global/director/service_apply.conf: 10:1-10:28) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'cluster-zone' (in /var/lib/icinga2/api/zones/director-global/director/service_apply.conf: 19:1-19:28) for type 'Service' does not match anywhere!
information/ConfigItem: Instantiated 1 ApiListener.
information/ConfigItem: Instantiated 6 Zones.
information/ConfigItem: Instantiated 4 Endpoints.
information/ConfigItem: Instantiated 2 ApiUsers.
information/ConfigItem: Instantiated 1 FileLogger.
information/ConfigItem: Instantiated 24 UserGroups.
information/ConfigItem: Instantiated 4 NotificationCommands.
information/ConfigItem: Instantiated 2440 Notifications.
information/ConfigItem: Instantiated 213 CheckCommands.
information/ConfigItem: Instantiated 1 IcingaApplication.
information/ConfigItem: Instantiated 618 Hosts.
information/ConfigItem: Instantiated 36 HostGroups.
information/ConfigItem: Instantiated 3 TimePeriods.
information/ConfigItem: Instantiated 3 Users.
information/ConfigItem: Instantiated 1 ServiceGroup.
information/ConfigItem: Instantiated 1598 Services.
information/ConfigItem: Instantiated 1 CheckerComponent.
information/ConfigItem: Instantiated 1 GraphiteWriter.
information/ConfigItem: Instantiated 1 IdoMysqlConnection.
information/ConfigItem: Instantiated 1 NotificationComponent.
information/ScriptGlobal: Dumping variables to file '/var/cache/icinga2/icinga2.vars'
information/cli: Finished validating the configuration file(s).
```
* If you run multiple Icinga 2 instances, the `zones.conf` file (or `icinga2 object list --type Endpoint` and `icinga2 object list --type Zone`) from all affected nodes. 

Primary Master (icinga2-master1.localdomain) config:
```
object Endpoint NodeName {
}

object Endpoint "icinga2-master2.localdomain" {
        host = "icinga2-master2.localdomain"
}

object Endpoint "icinga2-satellite1.localdomain" {
        host = "icinga2-satellite1.localdomain"
}

object Endpoint "icinga2-satellite2.localdomain" {
        host = "icinga2-satellite2.localdomain"
}

object Zone "master" {
        endpoints = [ NodeName, "icinga2-master2.localdomain" ]
}

object Zone "west" {
        endpoints = [ "icinga2-satellite1.localdomain" ]
        parent = "master"
}

object Zone "east" {
        endpoints = [ "icinga2-satellite2.localdomain" ]
        parent = "master"
}

object Zone "global-templates" {
        global = true
}

object Zone "director-global" {
        global = true
}
```
Secondary Master (icinga2-master2.localdomain) config:
```
object Endpoint NodeName {
}

object Endpoint "icinga2-master1.localdomain" {
        host = "icinga2-master1.localdomain"
}

object Endpoint "icinga2-satellite1.localdomain" {
        host = "icinga2-satellite1.localdomain"
}

object Endpoint "icinga2-satellite2.localdomain" {
        host = "icinga2-satellite2.localdomain"
}

object Zone "master" {
        endpoints = [ NodeName, "icinga2-master1.localdomain" ]
}

object Zone "west" {
        endpoints = [ "icinga2-satellite1.localdomain" ]
        parent = "master"
}

object Zone "east" {
        endpoints = [ "icinga2-satellite2.localdomain" ]
        parent = "master"
}

object Zone "global-templates" {
        global = true
}

object Zone "director-global" {
        global = true
}
```
Satellite 1 config:
```
object Endpoint NodeName {
}

object Endpoint "icinga2-master2.localdomain" {
}

object Endpoint "icinga2-master1.localdomain" {
}


object Zone "master" {
        endpoints = [ "icinga2-master1.localdomain", "icinga2-master2.localdomain" ]
}

object Zone "west" {
        endpoints = [ NodeName ]
        parent = "master"
}


object Zone "global-templates" {
        global = true
}

object Zone "director-global" {
        global = true
}
```
Satellite 2 config:
```
object Endpoint NodeName {
}

object Endpoint "icinga2-master2.localdomain" {
}

object Endpoint "icinga2-master1.localdomain" {
}


object Zone "master" {
        endpoints = [ "icinga2-master1.localdomain", "icinga2-master2.localdomain" ]
}

object Zone "east" {
        endpoints = [ NodeName ]
        parent = "master"
}


object Zone "global-templates" {
        global = true
}

object Zone "director-global" {
        global = true
}
```

**Comment by dnsmichi** [comment]
These commands being run on the secondary master just created a self-signed certificate with the local CA key pair. This is **different** to your primary master's CA and as such the TLS handshake fails, both instances do not trust each other.

You can verify that with `openssl s_client` and other variants mentioned in the troubleshooting docs. Simple fix - re-run node wizard on the second master as outlined in the distributed monitoring setup docs, and let it fetch a new signed certificate from the master.

**Comment by bl4kd3th** [comment]
@dnsmichi Thanks for your help. I figured out my exact issue. I ran the node wizard on the secondary master and set it up as a client/satellite and then immediately replaced the zones.conf with my desired configuration. This didn't allow the secondary master to connect to the primary to complete the certificate request.

Following your suggestion I wiped the director configs, reran the node wizard on the primary, and then ran the wizard on the other nodes. After they all connected and I completed the auto-sign process, I replaced the configs with my previous configs, reloaded icinga2 service, and they all worked as expected.

Thanks again for your help Michael!
